ui/src/app/cohort-search/modal/modal.component.tsx

```diff
@@ -0,0 +1,595 @@
+    height: auto;
+  }
+  @media (max-width: 1500px) {
+  .modal-container {
+    width: 90vw;
+  }
+  .modal-container.demographics {
+    width: 70vw;
+  }
+  @media (max-width: 1200px) {
+    .modal-container, .modal-container.demographics {
+      width: 100vw;
+    }
+  }
+  @media (max-height: 700px) {
+    .modal-container {
+      height: 100vh;
+    }
+  }
+  @media (max-height: 400px) {
+    .modal-container.demographics {
+      height: 100vh;
+    }
+    .modal-container.demographics.age {
+      height: auto;
+    }
+  }
+  @media (max-height: 300px) {
+    .modal-container.demographics.age {
+      height: 100vh;
+    }
+  }
+}
+`;
+
+function initGroup(role: string, item: any) {
+  return {
+    id: generateId(role),
+    items: [item],
+    count: null,
+    temporal: false,
+    mention: TemporalMention.ANYMENTION,
+    time: TemporalTime.DURINGSAMEENCOUNTERAS,
+    timeValue: '',
+    timeFrame: '',
+    isRequesting: false,
+    status: 'active'
+  };
+}
+
+interface Selection extends Criteria {
+  parameterId: string;
+}
+
+interface Props {
+  closeSearch: () => void;
+  searchContext: any;
+  setSearchContext: (context: any) => void;
+}
+
+interface State {
+  attributesNode: Criteria;
+  autocompleteSelection: Criteria;
+  backMode: string;
+  conceptType: string;
+  count: number;
+  disableFinish: boolean;
+  groupSelections: Array<number>;
+  hierarchyNode: Criteria;
+  loadingSubtree: boolean;
+  mode: string;
+  open: boolean;
+  selectedIds: Array<string>;
+  selections: Array<Selection>;
+  title: string;
+  treeSearchTerms: string;
+}
+
+export class CBModal extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      attributesNode: undefined,
+      autocompleteSelection: undefined,
+      backMode: 'list',
+      conceptType: null,
+      count: 0,
+      disableFinish: false,
+      groupSelections: [],
+      hierarchyNode: undefined,
+      loadingSubtree: false,
+      mode: 'list',
+      open: false,
+      selectedIds: [],
+      selections: [],
+      title: '',
+      treeSearchTerms: '',
+    };
+  }
+
+  componentDidMount(): void {
+    const {searchContext: {domain, item, standard, type}} = this.props;
+    if (!this.state.open) {
+      const selections = item.searchParameters;
+      const selectedIds = selections.map(s => s.parameterId);
+      if (type === CriteriaType.DECEASED) {
+        this.selectDeceased();
+      } else {
+        const title = domain === DomainType.PERSON ? typeToTitle(type) : domainToTitle(domain);
+        let {backMode, mode} = this.state;
+        let hierarchyNode;
+        if (this.initTree) {
+          hierarchyNode = {
+            domainId: domain,
+            type: type,
+            isStandard: standard,
+            id: 0,
+          };
+          backMode = 'tree';
+          mode = 'tree';
+        }
+        this.setState({backMode, hierarchyNode, mode, open: true, selectedIds, selections, title});
```

Comment:
Nice! I did not know we can do that 
this.setState({backMode, hierarchyNode i thought it was always like this.setState({backMode: backMode...

Reply:
Yeah, I like it better. Seems a little cleaner this way.